shared/lib_battery_dispatch.cpp

```diff
@@ -174,7 +174,7 @@ bool dispatch_t::check_constraints(double& I, size_t count)
     // Don't allow grid charging unless explicitly allowed (reduce charging)
     if (!m_batteryPower->canGridCharge && I < 0 && m_batteryPower->powerGridToBattery > tolerance)
     {
-        m_batteryPower->powerBatteryTarget += m_batteryPower->powerGridToBattery;
+        m_batteryPower->powerBatteryTarget += m_batteryPower->powerGridToBattery * m_batteryPower->singlePointEfficiencyACToDC;
```

Comment:
Are we going to handle different efficiencies in a future issue or pull request?

Reply:
Yes, if we do some of the DC-DC converter work we talked about with the PVSAMV1 refactor. Some of the DC-DC conversions are currently handled early in the powerflow code, the current setup should work as long as we keep using single point efficiencies.